### PR TITLE
[Model Compression] add v2 pruning examples

### PR DESCRIPTION
**Pull request by Fiascolsy**
### Description ###
These are simple test examples for all v2 pruning algorithms.
Add docstr 'scale' to basic_pruner.py

### Checklist ###
  - [ ] test case
  - [ ] doc

### How to test ###
Just run the command `python [NAME]_pruning_torch.py`

### CODE REVIEW

File: nni/algorithms/compression/v2/pytorch/pruning/basic_pruner.py [hunk 1]
```diff
@@ -379,6 +379,8 @@ def trainer(model: Module, optimizer: Optimizer, criterion: Callable[[Tensor, Te
         The criterion function used in trainer. Take model output and target value as input, and return the loss.
     training_epochs : int
         The epoch number for training model to sparsify the BN weight.
+    scale : float
+        Penalty parameters for sparsification.
```
Comment (QuanluZhang) [inline nni/algorithms/compression/v2/pytorch/pruning/basic_pruner.py:383]: better to explain more on this parameter. it is hard to understand with just four words

File: docs/en_US/Compression/v2_pruning_algo.rst [hunk 2]
```diff
@@ -266,6 +282,8 @@ Usage
    pruner = ADMMPruner(model, config_list, trainer, optimizer, criterion, iterations=10, training_epochs=1)
    masked_model, masks = pruner.compress()
 
+For detailed example please refer to :githublink:`examples/model_compress/pruning/v2/admm_pruning_example.py <examples/model_compress/pruning/v2/admm_pruning_example.py>`
+
```
Comment (QuanluZhang) [inline docs/en_US/Compression/v2_pruning_algo.rst:286]: i noticed that some pruners do not have corresponding example, why?

File: examples/model_compress/pruning/v2/activation_pruning_example.py [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+        for data, target in test_loader:
+            data, target = data.to(device), target.to(device)
+            output = model(data)
+            pred = output.argmax(dim=1, keepdim=True)
+            correct += pred.eq(target.view_as(pred)).sum().item()
+    acc = 100 * correct / len(test_loader.dataset)
+    print('Accuracy: {}%\n'.format(acc))
+    return acc
+
+def optimizer_schedular_generator(model, _lr=0.1, _momentum=0.9, _weight_decay=5e-4):
```
Comment (QuanluZhang) [inline examples/model_compress/pruning/v2/activation_pruning_example.py:70]: schedular -> scheduler

File: examples/model_compress/pruning/v2/level_pruning_example.py [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+    print('\n' + '=' * 50 + ' START TO PRUNE THE BEST ACCURACY PRETRAINED MODEL ' + '=' * 50)
+    config_list = [{
+        'sparsity': 0.5,
+        'op_types': ['default']
+    }]
+    pruner = LevelPruner(model, config_list)
+    _, masks = pruner.compress()
+    pruner.show_pruned_weights()
+    pruner._unwrap_model()
+    ModelSpeedup(model, dummy_input=torch.rand([10, 3, 32, 32]).to(device), masks_file=masks).speedup_model()
```
Comment (J-shang) [inline examples/model_compress/pruning/v2/level_pruning_example.py:111]: level pruner is a fine-grained pruner. recommended finetune with masks, do not speed up.

File: examples/model_compress/pruning/v2/slim_pruning_example.py [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+
+    # Start to prune and speedup
+    print('\n' + '=' * 50 + ' START TO PRUNE THE BEST ACCURACY PRETRAINED MODEL ' + '=' * 50)
+    config_list = [{
+            'total_sparsity': 0.5,
+            'op_types': ['BatchNorm2d'],
+    }]
+
+    optimizer, _ = optimizer_schedular_generator(model)
+    pruner = SlimPruner(model, config_list, trainer, optimizer, criterion, 1, 0.0001, 'normal')
```
Comment (J-shang) [inline examples/model_compress/pruning/v2/slim_pruning_example.py:110]: better to use this format to init, also in other example file:

`pruner = SlimPruner(model, config_list, trainer, optimizer, criterion, training_epochs=1, scale=0.0001, mode='global')`

and Slim is `global` mode in the paper, so we also better use `global` in the example.

File: docs/en_US/Compression/v2_pruning_algo.rst (hunk 2 above)
Comment (Fiascolsy) [reply docs/en_US/Compression/v2_pruning_algo.rst:286]: This PR was once planned to add basic pruning algorithms, but what you said was a good idea, I will add iterative pruning examples later : )

File: examples/model_compress/pruning/v2/activation_pruning_example.py (hunk 3 above)
Comment (Fiascolsy) [reply examples/model_compress/pruning/v2/activation_pruning_example.py:70]: got it

File: examples/model_compress/pruning/v2/slim_pruning_example.py (hunk 5 above)
Comment (Fiascolsy) [reply examples/model_compress/pruning/v2/slim_pruning_example.py:110]: understood

File: nni/algorithms/compression/v2/pytorch/pruning/basic_pruner.py (hunk 1 above)
Comment (Fiascolsy) [reply nni/algorithms/compression/v2/pytorch/pruning/basic_pruner.py:383]: I would fix this ASAP

File: examples/model_compress/pruning/v2/level_pruning_example.py (hunk 4 above)
Comment (Fiascolsy) [reply examples/model_compress/pruning/v2/level_pruning_example.py]: Both Level and ADMM pruner have been fixed.

**Review by QuanluZhang — APPROVED** [review]

**Review by J-shang — APPROVED** [review]
